vision.md

```diff
@@ -0,0 +1,407 @@
+    - As an example: Users wanting to set an SSL certificate should not need to
+      worry about also defining their DNS setup.
+1. Reproducability of a configuration
+1. Security
+    1. Operational Security
+        1. Components of the framework are encapsulated and do not influence
+           each others execution
+        1. The system can recover from bad states in a safe way
+        1. Untrusted input is handled in a way that does not influence execution
+           in a bad way (e.g. crashes the system)
+    1. Information Security
+        1. Data access is reglemented
+        1. Data origin is verified and potentially encrypted
+    1. Code Audit(ability)
+        1. Functionality can be audited easily because it is encapsulated (see
+           above)
+1. Generic solutions over specific solutions
+    - Problems are solved in a way that users can define the details as best
+      suited
+1. Easy configurability and high discoverability of configuration options
+    - Configuring the system is not done across dozens of configuration files at
+      different locations, and instead be centralized and easy to write
+    - Configuration options are easily discoverable and documented, for example,
+      in a highly secured environment, the user is not required to access some
+      online documentation ressource, but is able to access all configuration
+      documentation via the binary shipped or derived from the thin-edge.io
+      framework
+    - Configurations are representable in multiple ways, e.g. as text or in a
+      more visual format (graph, image overview)
+1. The user os able to enable and disable functionality of the framework
+    - Not all functionalities of the framework are required in every use-case,
+      users are able to disable or even remove those parts from deployed
+      binaries
+    - For example: if only approved and audited components are allowed to be
+      used
+1. Deriving specialized implementations/binaries from the framework is possible
+    - The thin-edge.io ships a potentially large all-in-one binary with as much
+      functionality as the project provides, but
+    - Tailoring down the binary and removing unused functionality is easily
+      doable with minimal effort by a System Integrator
+    - Building a specialized binary with additional custom functionality is
+      easily doable by an Plugin Developer
+1. As little overhead as possible
+    1. in CPU time: The Framework ensures that the absolute minimum of CPU time
+       is consumed to deliver the desired functionality the user configured the
+       framework to do
+    1. in Memory usage: The Framework ensures that as little as possible memory
+       is in use, at every point in execution time, as possible
+1. Compatibility
+    - Configuration and components are written in such a way that they may be
+      changed in the future with no or minimal impact on their intended purpose
+
+### Functional requirements
```

Comment:
Previously, we mention that thin-edge.io should also allow to integrate with existing projects / device software. I believe that this is important for each device manufacturer who has already significantly invested in device software and does not want to start from scratch. Is this a requirement we should add here?